src/core/lib/gprpp/windows/directory_reader.cc

```diff
@@ -60,20 +58,20 @@ std::unique_ptr<DirectoryReader> MakeDirectoryReader(
 // Reference for reading directory in Windows:
 // https://stackoverflow.com/questions/612097/how-can-i-get-the-list-of-files-in-a-directory-using-c-or-c
 // https://learn.microsoft.com/en-us/windows/win32/fileio/listing-the-files-in-a-directory
-absl::StatusOr<std::vector<std::string>>
-DirectoryReaderImpl::GetDirectoryContents() {
-  std::string search_path = absl::StrCat(directory_path_, "/*.*");
-  std::vector<std::string> files;
+absl::Status DirectoryReaderImpl::ForEach(
+    absl::FunctionRef<void(absl::string_view)> callback) {
+  // std::string search_path = absl::StrCat(directory_path_, "/*.*");
+  std::string search_path = absl::StrCat(directory_path_, "/*");
   WIN32_FIND_DATA find_data;
   HANDLE hFind = ::FindFirstFile(search_path.c_str(), &find_data);
   if (hFind != INVALID_HANDLE_VALUE) {
     do {
       if (!(find_data.dwFileAttributes & FILE_ATTRIBUTE_DIRECTORY)) {
-        files.push_back(std::string(find_data.cFileName));
+        callback(std::string(find_data.cFileName));
```

Comment:
Why are we constructing a `std::string` here?  The callback takes a `absl::string_view`.

Reply:
My mistake, at the time I was writing it I thought we'd have a problem implicitly converting from the `cFileName`